### PR TITLE
Rest Client Reactive: detect blocking the event loop

### PR DESCRIPTION
**Pull request by michalszynkiewicz**
very basic, run-time check

### CODE REVIEW

**Review by geoand — APPROVED** [review]

File: independent-projects/resteasy-reactive/client/runtime/src/main/java/org/jboss/resteasy/reactive/client/impl/InvocationBuilderImpl.java [hunk 1]
```diff
@@ -193,6 +194,11 @@ public Response get() {
     }
 
     private <T> T unwrap(CompletableFuture<T> c) {
+        if (Context.isOnEventLoopThread()) {
+            throw new IllegalStateException("Blocking rest client call made from the event loop. " +
```
Comment (gsmet) [inline independent-projects/resteasy-reactive/client/runtime/src/main/java/org/jboss/resteasy/reactive/client/impl/InvocationBuilderImpl.java:198]:
```suggestion
            throw new IllegalStateException("Blocking REST client call made from the event loop. " +
```

File: independent-projects/resteasy-reactive/client/runtime/src/main/java/org/jboss/resteasy/reactive/client/impl/InvocationBuilderImpl.java [hunk 2]
```diff
@@ -193,6 +194,11 @@ public Response get() {
...
+                    "If the code is executed from a RestEasy Reactive resource, either annotate the resource method " +
```
Comment (gsmet) [inline independent-projects/resteasy-reactive/client/runtime/src/main/java/org/jboss/resteasy/reactive/client/impl/InvocationBuilderImpl.java:199]:
```suggestion
                    "If the code is executed from a RESTEasy Reactive resource, either annotate the resource method " +
```

**Review by gsmet — COMMENTED** [review]
Spotted some case issues, feel free to ignore :).

@geoand should this be backported?

**Comment by geoand** [comment]
Yeah, this is really useful so we should backport it

**Pull request merged**